File: jeorg-camel-app-atm/src/main/java/com/jesperancinha/atm/finder/camel/AtmService.java
package com.jesperancinha.atm.finder.camel;

import com.jesperancinha.atm.finder.service.AtmLocatorServiceImpl;
import org.apache.camel.BeanInject;
import org.springframework.stereotype.Component;

import java.io.IOException;

@Component
public class AtmService {

    @BeanInject
    private AtmLocatorServiceImpl atmLocatorService;

    public AtmProvider getATMProvider(String city) throws IOException {
        return AtmProvider.builder().atmMachines(atmLocatorService.getAtmPerCity(city.toUpperCase()))
                .city(city).build();
    }
}

File: jeorg-camel-app-atm/src/main/kotlin/com/jesperancinha/atm/finder/camel/AtmService.kt
package com.jesperancinha.atm.finder.camel

import com.jesperancinha.atm.finder.service.AtmLocatorServiceImpl
import org.apache.camel.BeanInject
import org.springframework.stereotype.Component
import java.io.IOException
import java.util.*

@Component
class AtmService {
    @BeanInject
    private val atmLocatorService: AtmLocatorServiceImpl? = null
    @Throws(IOException::class)
    fun getATMProvider(city: String): AtmProvider {
        return AtmProvider.builder().atmMachines(atmLocatorService!!.getAtmPerCity(city.uppercase(Locale.getDefault())))
            .city(city).build()
    }
}